src/python/pants/option/global_options.py

```diff
@@ -415,6 +421,20 @@ def register_bootstrap_options(cls, register):
     register('--process-execution-cleanup-local-dirs', type=bool, default=True, advanced=True,
              help='Whether or not to cleanup directories used for local process execution '
                   '(primarily useful for e.g. debugging).')
+    register('--process-execution-speculation-timeout-ms', type=int,
+             default=DEFAULT_EXECUTION_OPTIONS.process_execution_speculation_timeout_ms, advanced=True,
+             help='Number of milliseconds to wait before speculating a second request for a slow process. '
```

Comment:
Was there discussion already around seconds vs. milliseconds? I find seconds easier to reason about, e.g. 0.1 seconds is more intuitive to me than 100 ms.

I imagine many `ExecuteProcessRequest`s will happen quickly, but for things like Pytest they can take up to a few minutes. Of course, we're not trying to run both processes to completion so we should not encourage users to use a timeout like `60 seconds`, but with Pytest running unit tests something like `0.5 s` or `1 s` would likely be more appropriate, and that's easier to express in seconds than `500 ms` and `1000 ms` imo.

Reply:
I generally agree that we should be consistent with time units across our CLI, and seconds does seems to be the most common time unit. For this flag I think the long term goal is to have the speculation timeout inferred from a distribution of timings for a specific operation, but for now just expose *something*.

>Of course, we're not trying to run both processes to completion so we should not encourage users to use a timeout like 60 seconds

I think there is some confusion here, maybe the flag is a misnomer. This is the delay to wait before executing a second request, not the timeout on the second request itself. would the name `--process-execution-speculation-delay-ms` be more intuitive?

Reply:
> I generally agree that we should be consistent with time units across our CLI, and seconds does seems to be the most common time unit.

Agreed. For example, `pantsd` uses seconds. Definitely think there's value in using seconds, given that you can use decimals for more precision.

> I think there is some confusion here, maybe the flag is a misnomer. This is the delay to wait before executing a second request, not the timeout on the second request itself. would the name --process-execution-speculation-delay-ms be more intuitive?

Oh huh, maybe I don't understand speculation as well as I thought! My understanding was that you first try to see if it passes locally within n seconds. If it doesn't, abort and run via remote execution.

Are you saying that after the threshold, you will have both a local and a remote process running? And then whichever finishes first finishes first and cancels the other? If this is the case, the new flag is a more clear name. 

(Also, if this is the case, we already struggle with out of memory for V2 unit tests in Travis. Would this exacerbate the risk of OOM?)

Reply:
Yes, we let the two requests race each other, and take the first result. The speculation tests mock out any RPCs or subprocesses so they wont add much memory overhead. An integration test which uses speculation might, but only if remote_execution is enabled.